pkg/cloudprovider/vsphere/config.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"gopkg.in/gcfg.v1"
+)
+
+// CPIConfig is used to read and store information (related only to the CPI) from the cloud configuration file
+type CPIConfig struct {
```

Comment:
Thanks for the update on this PR! I think we might be just about there. We really need to carefully think about the config and thinking out loud here (not sure if this is a good idea or not), maybe we might be able to do something to the effect of...

```
type CPIConfig struct {
  vcfg.Config

  CPI struct {
    // ... comments ...
    InternalNetworkSubnetCIDR string `gcfg:"internal-network-subnet-cidr"`
    ExternalNetworkSubnetCIDR string `gcfg:"external-network-subnet-cidr"`
  }
}
```

That way the shared config is still a single configuration object versus having two separate config objects (one which is shared and a separate CPI one). CSI specific fields would then go into a `CSI struct` similar to this CPI.

Reply:
So this is going to add a `[CPI]` section to the config file? Wondering if something more descriptpive is useful to users, maybe `[Nodes]`  or something else

Reply:
Yes, it would add a `[CPI]` section... yea I think something more descriptive might be better. I could definitely get behind something like `[Nodes]`.

Reply:
I think it should be ok... But just need to float the idea out there that if we call it something like `[Nodes]` on the CSI side they could come up with a similar section, have a parameter under that section, but that parameter has a completely different meaning than the CPI version of it.

Reply:
The config parsing does support subsections... we could do `[CPI "Nodes"] (then CSI could do [CSI "Nodes"])`

Reply:
I discussed this with @andrewsykim and the CSI repo maintainers and the consensus is to use `[Nodes]` to contain:
```
InternalNetworkSubnetCIDR string `gcfg:"internal-network-subnet-cidr"`
ExternalNetworkSubnetCIDR string `gcfg:"external-network-subnet-cidr"`
```

Reply:
It is done. I've also added tests for the new config.